pluginframework/resource_quality_monitor.go

```diff
@@ -150,15 +168,21 @@ func (r *QualityMonitorResource) Update(ctx context.Context, req resource.Update
 		resp.Diagnostics.AddError("Failed to get workspace client", err.Error())
 		return
 	}
-	var updateMonitorTfSDK catalog_tf.UpdateMonitor
-	diags := req.Plan.Get(ctx, &updateMonitorTfSDK)
+	var monitorInfoTfSDK catalog_tf.MonitorInfo
+	diags := req.Plan.Get(ctx, &monitorInfoTfSDK)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	// Plan is not adding `dashboard_id`, but it is in the state.
+	diags = req.State.GetAttribute(ctx, path.Root("dashboard_id"), &monitorInfoTfSDK.DashboardId)
```

Comment:
Not sure why this is the case, it seems like we need to manually add it as it is only in the state